TDL-06-EC2-Cloudwatch-S3/terraform-code-ec2-cw-s3/ec2.tf

```diff
@@ -0,0 +1,21 @@
+resource "aws_instance" "instance" {
+  ami                         = var.AMIS[var.aws_region]
```

Comment:
It is better to avoid hardcoded instance_type,  key_name:) what about 
instance_type = var.instance_type
key_name        = var.key_name

and set the default value for instance_type & key_name in ur variable.tf so you can make ur code re-usable?